packages/dev/sharedUiComponents/src/fluent/primitives/colorPicker.tsx

```diff
@@ -0,0 +1,407 @@
+        display: "flex",
+        alignItems: "flex-end",
+        flexDirection: "row",
+        gap: "10px",
+    },
+    colorFieldWrapper: {
+        display: "flex",
+        flexDirection: "column",
+        gap: "6px",
+    },
+    input: {
+        width: "80px",
+    },
+    spinButton: {
+        minWidth: "60px",
+    },
+    container: {
+        display: "flex",
+        flexDirection: "column",
+        gap: "16px",
+    },
+    row: {
+        display: "flex",
+        flexDirection: "row",
+        gap: "12px",
+        alignItems: "center",
+    },
+});
+
+export const ColorPickerPopup = (props: ColorPickerProps<Color3 | Color4>) => {
+    const hexId = useId("hex-input");
+    const alphaId = useId("alpha-input");
+
+    const classes = useColorPickerStyles();
+    const [color, setColor] = React.useState(props.value);
+
+    const [popoverOpen, setPopoverOpen] = React.useState(false);
+
+    React.useEffect(() => {
+        props.onChange(color);
+    }, [color]);
+
+    const handleChange: FluentColorPickerProps["onColorChange"] = (_, data) => {
+        let color: Color3 | Color4 = Color3.FromHSV(data.color.h, data.color.s, data.color.v);
+        if (props.value instanceof Color4) {
+            color = Color4.FromColor3(color, data.color.a ?? 1);
+        }
+        setColor(color);
+    };
+
+    const onRgbChange: InputRgbFieldProps["onChange"] = (_, data) => {
+        if (data.value) {
+            const newColor = color.clone();
+            newColor[data.rgbKey] = data.value;
+            setColor(newColor);
+        }
+    };
+
+    const onHsvChange: InputHsvFieldProps["onChange"] = (_, data) => {
+        if (data.value) {
+            // Convert current color to HSV, update the new hsv value, then update state
+            const hsv = rgbaToHsv(color);
+            hsv[data.hsvKey] = data.value;
+            let newColor: Color3 | Color4 = Color3.FromHSV(hsv.h, hsv.s, hsv.v);
+            if (color instanceof Color4) {
+                newColor = Color4.FromColor3(newColor, color.a ?? 1);
+            }
+            setColor(newColor);
+        }
+    };
+
+    const onAlphaChange = React.useCallback(
+        (_ev: SpinButtonChangeEvent, data: SpinButtonOnChangeData) => {
+            const value = data.value ?? parseFloat(data.displayValue ?? "");
+
+            if (Number.isNaN(value) || value < 0 || value > 1) {
+                return;
+            }
+
+            let newColor = color;
+            if (props.value instanceof Color4) {
+                newColor = Color4.FromColor3(color, value ?? 1);
+            }
+            setColor(newColor);
+        },
+        [setColor, color]
+    );
+
+    const renderGammaLinearInfo = () => {
+        return (
+            <InfoLabel
+                info={
+                    <>
+                        This color picker is attached to a material whose color is stored in linear space (ex: PBR Material), and Babylon converts the color to gamma space before

```

Comment:
Seems like this is indicating the color picker is specifically for materials, but presumably we'd want to use it for other color cases like the scene clear color?